src/components/ebay-phone-input/component.ts

```diff
@@ -0,0 +1,132 @@
+import type { WithNormalizedProps } from "../../global";
+import type { Input as MenuButtonInput } from "../ebay-menu-button/component";
+import type { default as Textbox } from "../ebay-textbox/component-browser";
+import mask, { stripNonDigits } from "../../common/mask";
+
+import countries, {
+    type CountriesLanguageInterface,
+    type CountryInterface,
+    countriesEnglish,
+} from "../../common/countries";
+import { AttrString } from "marko/tags-html";
+
+export interface CountryNames {
+    [index: string]: string;
+}
+
+export interface PhoneInputEvent {
+    originalEvent?: Event;
+    value?: string;
+    rawValue?: string;
+    callingCode?: string;
+    countryCode?: string;
+}
+
+interface PhoneInputInput extends Omit<MenuButtonInput, `on${string}`> {
+    disabled?: boolean;
+    "country-names"?: CountriesLanguageInterface;
+    "a11y-button-text"?: AttrString;
+    "a11y-menu-text"?: MenuButtonInput["a11yText"];
+    "floating-label"?: string;
+    readonly?: boolean;
+    invalid?: boolean;
+    "country-code"?: string;
+    value?: string;
+    "on-keyup"?: (event: PhoneInputEvent) => void;
+    "on-keypress"?: (event: PhoneInputEvent) => void;
+    "on-keydown"?: (event: PhoneInputEvent) => void;
+    "on-focus"?: (event: PhoneInputEvent) => void;
+    "on-blur"?: (event: PhoneInputEvent) => void;
+    "on-collapse"?: (event: PhoneInputEvent) => void;
+    "on-expand"?: (event: PhoneInputEvent) => void;
+    "on-change"?: (event: PhoneInputEvent) => void;
+}
+
+export interface Input extends WithNormalizedProps<PhoneInputInput> {}
+
+export interface State {
+    index: number;
+}
+
+class PhoneInput extends Marko.Component<Input, State> {
+    mask: any;
+
+    onCreate() {
+        this.state = {
+            index: 0,
+        };
+    }
+
+    get selectedCountry(): CountryInterface {
+        const countryNames = this.input.countryNames || countriesEnglish;
+        const currentCountryName = countryNames[this.state.index];
+        return countries[currentCountryName[1]];
+    }
+
+    onInput(input: Input) {
+        const countryNames: CountriesLanguageInterface =
+            input.countryNames || countriesEnglish;
+        const { countryCode } = input;
+        if (countryCode) {
+            let index = countryNames.findIndex(
+                (country) => country[1] === countryCode.toUpperCase(),
+            );
+            if (index === -1) {
+                index = 0;
+            }
+            this.state.index = index;
+        }
+    }
+    handleCountryChange(e: any) {
+        this.state.index = e.index;
+        // Trigger change event after state is updated so that the right mask value will be emitted
+        setTimeout(() => {
+            this.forwardEvent("change", { originalEvent: e });
+        }, 0);
+    }
+
+    forwardEvent(eventName: string, event: PhoneInputEvent) {
+        const selectedCountry = this.selectedCountry;
+
+        setTimeout(() => {
```

Comment:
Why are doing this `setTimeout`?

Reply:
```
// Trigger change event after state is updated so that the right mask value will be emitted
 ```
The stack needs to be cleared so that the state is updated and then I pull the right value from the mask. 
This was broken with Imask, but might need to recheck for Michaels version.

Reply:
Yes this is still needed. I removed one of the timeouts, but the other one is needed.

What happens is the input doesn't get the masking applied from the new dropdown selection if setTimeout is removed. 
Try entering a phone number, and then change the value in the listbox
If we remove setTimoue the old mask is applied.